lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py

```diff
@@ -217,238 +273,109 @@ def test_clear_tools_test70(monkeypatch, agent_config, pbench_run, pbench_cfg):
-    monkeypatch.setenv("_PBENCH_AGENT_CONFIG", str(pbench_cfg))
-    default_group = pbench_run / "tools-v1-default"
-    default_group.mkdir(parents=True)
-    fubar_default_host = default_group / "fubar6.example.com"
-    fubar_default_host.mkdir(parents=True, exist_ok=True)
-    pidstat_default_tool = fubar_default_host / "pidstat"
-    pidstat_default_tool.touch()
-
-    another_group = pbench_run / "tools-v1-another"
-    another_group.mkdir(parents=True)
-    fubar_another_host = another_group / "fubar7.example.com"
-    fubar_another_host.mkdir(parents=True, exist_ok=True)
-    pidstat_another_tool = fubar_another_host / "pidstat"
-    pidstat_another_tool.touch()
-
-    command = ["pbench-clear-tools", "--groups=default,another"]
-    out, err, exitcode = pytest.helpers.capture(command)
-    assert b"" == out
-    assert default_group.exists() is True
-    assert another_group.exists() is False
-
-
-def test_clear_tools_test72(monkeypatch, agent_config, pbench_run, pbench_cfg):
-    """ Attempt to clear the tool groups, by name, separated by comma.
-    However one of the tool group name is wrong"""
-    monkeypatch.setenv("_PBENCH_AGENT_CONFIG", str(pbench_cfg))
-
-    another_group = pbench_run / "tools-v1-another"
-    another_group.mkdir(parents=True)
-    fubar_another_host = another_group / "fubar.example.com"
-    fubar_another_host.mkdir(parents=True, exist_ok=True)
-    pidstat_another_tool = fubar_another_host / "pidstat"
-    pidstat_another_tool.touch()
-
-    command = ["pbench-clear-tools", "--groups=another,wrong"]
-    out, err, exitcode = pytest.helpers.capture(command)
-    assert b"" == out
-    assert (
-        b'Removed "pidstat" from host "fubar.example.com" in tools group '
-        b'"another"' in err
-    )
-    assert b'pbench-clear-tools: invalid --group option "wrong"' in err
-    assert another_group.exists() is False
-
-
 @pytest.mark.parametrize(
-    "groups", ["default", "default,custom1", "default,custom1,custom2"]
+    "groups",
+    ["", "default", "default,group1", "default,group1,group2", "group1,group2,group3",],
 )
 @pytest.mark.parametrize(
     "remotes",
-    ["remote1", "remote1,remote2", "remote1,remote2,remote3", "remote2,remote4"],
+    [
+        "",
+        "remote1.example.com",
+        "remote1.example.com,remote2.example.com",
+        "remote1.example.com,remote2.example.com,remote3.example.com",
+        "remote2.example.com,remote4.example.com",
+    ],
 )
 @pytest.mark.parametrize(
-    "tools", ["pidstat", "pidstat,mpstat", "pidstat,mpstat,iostat", "mpstat," "vmstat"]
+    "tools",
+    ["", "pidstat", "pidstat,mpstat", "pidstat,mpstat,iostat", "mpstat,vmstat"],
 )
-def test_clear_tools_test73(
-    monkeypatch, tmp_path, agent_config, pbench_run, pbench_cfg, groups, remotes, tools
+def test_clear_tools_test85(
+    monkeypatch,
+    tmp_path,
+    agent_config,
+    pbench_run,
+    pbench_cfg,
+    tools_configuration,
```

Comment:
Note that since `tools_configuration` already references `pbench_run` and `pbench_cfg` you don't need to repeat them here unless you need the values in this method ... which so far as I can see you don't. A shorter list would be easier to read. It's possible some of the others are also nested fixtures that could be omitted. (Although it may not matter much unless you can get it down to one line.)